jupyter_book/page/page.py

```diff
@@ -0,0 +1,267 @@
+    clear_output: bool
+        To remove the output from notebook
+
+    Returns
+    =======
+    page : HTML document
+        The input content file converted to HTML format.
+    """
+
+    if preprocessors is None:
+        preprocessors = []
+    elif not isinstance(preprocessors, (list, tuple)):
+        preprocessors = [preprocessors]
+
+    if name is None:
+        name = "notebook"
+
+    ########################################
+    # Notebook cleaning
+    _clean_markdown_cells(ntbk)
+
+    #############################################
+    # Preprocessor configuration
+    c = Config()
+
+    # Remove cell elements using tags
+    c.TagRemovePreprocessor.remove_cell_tags = ("remove_cell", "removecell")
+    c.TagRemovePreprocessor.remove_all_outputs_tags = ("remove_output",)
+    c.TagRemovePreprocessor.remove_input_tags = ("remove_input",)
+
+    # Remove any cells that are *only* whitespace
+    c.RegexRemovePreprocessor.patterns = ["\\s*\\Z"]
+
+    c.HTMLExporter.preprocessors = [
+        "nbconvert.preprocessors.TagRemovePreprocessor",
+        "nbconvert.preprocessors.RegexRemovePreprocessor",
+    ]
+
+    if clear_output:
+        c.HTMLExporter.preprocessors.append('nbconvert.preprocessors.ClearOutputPreprocessor')
+
+    if path_media_output is not None:
+        # So the images are written to disk
+        c.HTMLExporter.preprocessors.append(
+            "nbconvert.preprocessors.ExtractOutputPreprocessor"
+        )
+
+    # Add extra preprocessors given by the user
+    for preprocessor in preprocessors:
+        c.HTMLExporter.preprocessors.append(preprocessor)
+
+    # The text used as the text for anchor links.
+    # The text used as the text for anchor links.
+    # TEMPORATILY Set to empty since we'll use anchor.js for the links
+    # Once https://github.com/jupyter/nbconvert/pull/1101 is fixed
+    # set to '<i class="fas fa-link"> </i>'
+    c.HTMLExporter.anchor_link_text = " "
+
+    # Excluding input/output prompts
+    c.HTMLExporter.exclude_input_prompt = True
+    c.HTMLExporter.exclude_output_prompt = True
+
+    #############################################
+    # Run and convert to HTML
+
+    # Excution of the notebook if we wish
+    if execute_dir is not None:
+        ntbk = run_ntbk(ntbk, execute_dir, allow_errors=True)
+
+    # Generate HTML from our notebook using the template
+    output_resources = {"output_files_dir": path_media_output, "unique_key": name}
+    exp = HTMLExporter(template_file=PATH_TEMPLATE, config=c)
+    html, resources = exp.from_notebook_node(ntbk, resources=output_resources)
+
+    # Add title and author information if it's provided
+    title = '' if title is None else f'<div id="page-title">{title}</div>'
+    author = '' if author is None else f'<div id="page-author">{author}</div>'
+
+    html = f"""
+    <main class="jupyter-page">
+    {title}
+    {author}
+    {html}
+    </main>
+    """
+    return html, resources
+
+
+def page_head():
```

Comment:
Would it be worthwhile to also let users add in their own JS / CSS here, as we do in the full book ?

Reply:
Ah that's a good idea - lemme give it a shot